### ISSUE TITLE
Daily limit exceeded

### ISSUE THREAD

**Issue by maridansoft** [bromix/plugin.video.youtube#343]
Hi,
can you tell how to fix it? Where can I set my own API key and secret???

**Comment by bromix** [comment]
At the moment not possible.

**Comment by maridansoft** [comment]
what you mean? all are using the same API , no? I want to make my own api and set it up, it's not possible?

**Comment by maridansoft** [comment]
Isn't it this one? 
https://github.com/bromix/plugin.video.youtube/blob/5dc877e5a56fe0e2dac5b0595afbc5fbbfbaf53d/resources/lib/youtube/client/login_client.py

**Comment by bromix** [comment]
Yes, it is..what I was saying is, it's not available via settings...if, you've to hard-code it.

**Comment by jimporter** [comment]
I'm guessing the idea behind the multiple config entries is to reduce the likelihood of hitting API rate limits, right? Would it make more sense to use something else as the key into the `CONFIGS` dict instead of the Kodi version? For instance, you could take a hash of the user's MAC address when they log in to pick out a key and then store that key in the add-on's settings for later use. Then you could have as many different API keys as you need to avoid hitting the rate limit.

**Comment by bromix** [comment]
You're right with the API keys based on the version of KODI...but it's not enough anymore :)

I've another idea which should work better for each one...but my time very limited at the moment to implement the solution.

**Comment by joehanchoi** [comment]
I've since just forked your project and edited the keys locally. I'd obviously prefer staying on your build as it means not having to merge changes. Take your time, though.

**Comment by clarjon1** [comment]
@joehanchoi How did you setup your api keys? I keep getting a blank screen when I try to go to my subscriptions, so i'm clearly not setting -something- right.
